libvast/test/format/json.cpp

```diff
@@ -121,8 +121,10 @@ TEST(json hex number parser) {
   CHECK_EQUAL(x, -123.0);
   CHECK(json_number("123", x));
   CHECK_EQUAL(x, 123.0);
-  CHECK(json_number("+123", x));
-  CHECK_EQUAL(x, 123.0);
+  // JSON does not allow `+` before a number.
```

Comment:
Does this matter if it occurs in real data? For parsing, we should always follow the principle "be lenient what you accept, and strict what you emit." Users typically don't care if their tool only conforms 99% to a spec.

Reply:
> we should always follow the principle "be lenient what you accept, and strict what you emit."

I agree with that, but does it _actually_ appear in _real_ data (as in not so niche it doesn't exist from a practical point of view).

I think you need to bring evidence for the need first, otherwise it isn't wise to spend cycles on this.

Reply:
Agreeing here. It looked like it used to work, but now does no longer. If this indeed takes a non-trivial amount of cycles to bring back, then ignore it.